pkg/rtc/mediatrackreceiver.go

```diff
@@ -151,6 +168,36 @@ func (t *MediaTrackReceiver) SetupReceiver(receiver sfu.TrackReceiver, priority
 	t.MediaTrackSubscriptions.Start()
 }
 
+func (t *MediaTrackReceiver) SetPotentialCodecs(codecs []webrtc.RTPCodecParameters, headers []webrtc.RTPHeaderExtensionParameter) {
+	t.lock.Lock()
+	t.potentialCodecs = codecs
+	for i, c := range codecs {
+		var exist bool
+		for _, r := range t.receivers {
+			if strings.EqualFold(c.MimeType, r.Codec().MimeType) {
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			t.receivers = append(t.receivers, &simulcastReceiver{
+				TrackReceiver: NewDumbReceiver(livekit.TrackID(t.trackInfo.Sid), string(t.PublisherID()), c, headers),
+				priority:      i,
+			})
+		}
+	}
+	sort.Slice(t.receivers, func(i, j int) bool {
+		return t.receivers[i].Priority() < t.receivers[j].Priority()
+	})
+	t.shadowReceivers()
```

Comment:
Suggest renaming to `shadowReceiversLocked` to follow convention from other places.